design/datamover.md

```diff
@@ -0,0 +1,208 @@
+   plugin would lookup for the PVCs present in the user namespace and will create a DMB CR for every PVC. This Plugin is also responsible for creating DataMoverRestore(DMR) CR for each DMB CR encountered during restore.
+
+6. DataMoverBackup Controller: The DataMoverBackup Controller will watch for DataMoverBackup CR.
+
+7. DataMoverRestore Controller: DataMoverRestore Controller will watch for DataMoverRestore CR.
+
+**Note**: Currently, we have taken the DataMoverRestore Controller approach but we want to explore upstream K8s APIs for these usecases like [volume populator](https://kubernetes.io/blog/2021/08/30/volume-populators-redesigned/).
+
+### Data Mover controller PoC - [VolSync](https://github.com/backube/volsync)
+
+VolSync will be used as the Data Mover and `restic` (VolSync restic option and not the one integrated with Velero) will be the supported method for backup & restore of PVCs. Restic repository details are configured in a `secret` object
+which gets used by the VolSync's resources. This design takes advantage of VolSync's two resources - `ReplicationSource` & `ReplicationDestination`. `ReplicationSource` object helps
+with taking a backup of the PVCs and using restic to move it to the storage specified in the restic secret. `ReplicationDestination` object takes care of restoring the backup from the
+restic repository. There will be a 1:1 relationship between the replication src/dest CRs and PVCs.
+
+Here, the user will create a restic secret. Using that secret as source, the controller will create on-demand secrets for every backup/restore request.
+
+Assuming that the DataMover flag is enabled, then the user needs to create a restic secret with all the following details (this should be done before velero install just like `cloud-credentials`),
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: restic-config
+type: Opaque
+stringData:
+  # The repository url
+  RESTIC_REPOSITORY: s3:s3.amazonaws.com/<bucket>
+  # The repository encryption key
+  RESTIC_PASSWORD: <password>
+  # ENV vars specific to the chosen back end
+  # https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html
+  AWS_ACCESS_KEY_ID: <access_id>
+  AWS_SECRET_ACCESS_KEY: <access_key>
+```
+*Note: More details for installing restic secret in [here](https://volsync.readthedocs.io/en/stable/usage/restic/index.html#specifying-a-repository)*
+
+
+Like any other plugins, DataMover plugin gets deployed as an init container in the velero deployment. This plugin will be responsible for creating `DataMoverBackup` & `DataMoverRestore` CRs.
+
+Once a DataMoverBackup CR gets created, the controller will create the corresponding `ReplicationSource` CR in the protected namespace. VolSync watches for the creation of `ReplicationSource` CR and copies the PVC data to the restic repository mentioned in the `restic-secret`.
```

Comment:
By reading the following section I don't quite understand how CSI volumesnapshotcontent is moved.